### ISSUE TITLE
Support for leqno and fleqn rendering options

### ISSUE THREAD

**Issue by willymcallister** [KaTeX/KaTeX#1776]
In katex 0.10+ is it possible to render a left-justified equation on its own line and include a \tag{} or \tag*{}?

**Comment by edemaine** [comment]
Thanks! I'd been meaning to add an issue for this.

The fix for now is to add this CSS to your page:
```css
.katex-display > .katex > .katex-html > .tag !important { left: 0; right: auto; }
```

We should add this to the documentation, or add an `leqno` option to the options list. (In LaTeX, this is triggered via `\usepackage[leqno]{amsmath}`.)

**Comment by willymcallister** [comment]
I added this at the end of katex.css (same as you suggested, without the "!important").

```
.katex-display > .katex > .katex-html > .tag { 
  left: 0; 
  right: auto; 
}

```
The corresponding html in my article.md file is, 

```
\\[\tag{lo} z = 3\\]

\\[\tag{hi} x+y^{2x}\\]
```

This produces a centered equation with the tag on the left, as shown in this screenshot,
<img width="735" alt="screen shot 2018-11-05 at 1 58 33 pm" src="https://user-images.githubusercontent.com/17035787/48029393-f2884080-e102-11e8-9f23-ab4a43829451.png">

My goal was to have the equation on the left (not centered), and the tag on the far right. I did this by removing the css above and changing one line in katex.css, line 966,

994 .katex-display > .katex {
995   display: block;
996   text-align: left;     //changed from center to left, WMc
997   white-space: nowrap;
998 }

Achieving the goal,
<img width="714" alt="screen shot 2018-11-05 at 2 04 05 pm" src="https://user-images.githubusercontent.com/17035787/48029776-f23c7500-e103-11e8-9261-ce7868b08310.png">

**Comment by edemaine** [comment]
Oops, I misread "left-justified equation" as "left-justified tag" and answered the latter.  That is indeed how you'd do left-justified equations.  Is there a way to do this in LaTeX?  (If so, we could add explicit support in KaTeX, but if not, we probably won't.)

**Comment by willymcallister** [comment]
The amsmath package has a command `fleqn`, "flush left equation". This puts display equations on the left rather than centered. But: `fleqn` may be a mode for all equations rather than per-equation. There is also an `flalign` that seems similar but probably with different side effects. You have much deeper understanding of the arcanery within Latex.

**Comment by edemaine** [comment]
Cool, yes, I believe KaTeX should support both `fleqn` and `leqno` global options to match LaTeX's amsmath, so I've renamed this issue accordingly.

`flalign` is tougher in the current KaTeX model, because it applies only at the top text level, not within math.  It could be done as part of #604 though.

In LaTeX, it is possible to change the global `fleqn` setting, but it is messy. With `\makeatletter` on, left justification can be turned on via `\@mathmargin30pt\relax`, and centering can be turned on via `\@mathmargin\@centering`. (I got this by looking at the amsmath source code.) This would be somewhat difficult to implement in KaTeX, but I don't think we need to support this level of configurability.
